modules/systemd/boot.nix

```diff
@@ -0,0 +1,109 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+
+  # Ghaf configuration flag
+  cfg = config.ghaf.systemd.boot;
+
+  # Import minimal systemd config
+  minimalSystemd = (import ./minimal.nix { inherit pkgs; }).minimalSystemd;
+
+  # Initrd-systemd package configuration 
+  bootPackageConfiguration =  minimalSystemd.override {
+    pname = "boot-systemd";
+    withAcl = true;
+    withCompression = true;
+    withCoredump = true;
+    withCryptsetup = true;
+    withEfi = pkgs.stdenv.hostPlatform.isEfi;
+    withFido2 = true;
+    withKmod = true;
+    withTpm2Tss = true;
+  };
+    
+  # Excluded systemd units for initrd
+  excludedInitrdUnits = [
+      
+    # Targets
+    # "basic.target"
+    # "swap.target"
+    # "initrd.target"
+    # "initrd-fs.target"
+    # "initrd-switch-root.target"
+    # "initrd-root-device.target"
+    # "initrd-root-fs.target"
+    # "sysinit.target"
+    (if config.ghaf.profiles.debug.enable then "emergency.target" else "")    
+    "paths.target"
+    "slices.target"
+    "sockets.target"
+    # "final.target"
+    "halt.target"
+    "local-fs-pre.target"
+    "local-fs.target"
+    # "poweroff.target"
+    "reboot.target"
+    "rescue.target"
+    "shutdown.target"
+    "multi-user.target"         
+    "ctrl-alt-del.target"
+    "timers.target"
+    # "umount.target"
+    "kexec.target"
+
+    # Services
+    # "initrd-cleanup.service"
+    # "initrd-switch-root.service"
+    # "initrd-parse-etc.service"
+    (if config.ghaf.profiles.debug.enable then "emergency.service" else "")    
+    "kmod-static-nodes.service"
+    "sys-kernel-config.mount"
+    "systemd-modules-load.service"        
+    "systemd-halt.service"
+    "sigpwr.target"
+    "rescue.service"
+    "systemd-journald.service"
+    "systemd-poweroff.service"
+    "systemd-reboot.service"
+    "systemd-sysctl.service"
+    "systemd-ask-password-console.service"
```

Comment:
Would 
`(if config.ghaf.profiles.debug.enable then "systemd-ask-password-console.service" else "")`
work here?

Reply:
Same as above for the options. My approach here was to disable all shell access except for emergency access in debug, as in stage 1 there should be no need to access other than debugging. For the host configuration, something along the lines of what you suggested should be implemented though, also keeping in mind the host only configuration. As of now, I only excluded `getty.target ` as I think this was causing the glitch in vm configuration...